pkg/file/lazy_bounded_read_closer.go

```diff
@@ -80,10 +84,10 @@ func (d *lazyBoundedReadCloser) ReadAt(b []byte, off int64) (n int, err error) {
 
 	n, err = d.reader.ReadAt(b, off)
 	if err != nil && errors.Is(err, io.EOF) {
-		// we've reached the end of the file, force a release of the file descriptor. If the file has already been
-		// closed, ignore the error.
-		if closeErr := d.file.Close(); !errors.Is(closeErr, os.ErrClosed) {
-			return n, closeErr
+		// we've reached the end of the file, release of the file descriptor. continue to return EOF
+		// IMPORTANT: call d.Close to reset internal state
+		if closeErr := d.Close(); closeErr != nil {
+			log.Tracef("unable to close: %v: %v", d.path, closeErr)
```

Comment:
I'm not entirely sure the best behavior here, can certainly be convinced to change this... but this is only hit if we received EOF during the Read call, which will be returned to the caller and we're managing open file handles behind the scenes, so while an error closing here can indicate a problem, there's not much that a call can do about this and they are already receiving an EOF, so should stop further reading anyway unless they Seek, at which time we need to be able to reopen the file